File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        boolean isSubsequence = false;
        int next = 0;
        if(s.isEmpty()) return true;

        for(int i = 0; i < s.length(); i++){
            char subChar = s.charAt(i);
            isSubsequence = false;
            
            for(int j = next; j < t.length(); j++){
                if(t.charAt(j) == subChar){
                    isSubsequence = true;
                    next = j + 1;
                    break;
                }
            }
            if(!isSubsequence) break;
        }
        return isSubsequence;
    }
}

File: 0392-is-subsequence/0392-is-subsequence.kt
class Solution {
    fun isSubsequence(s: String, t: String): Boolean {
        val sLength = s.length
        val tLength = t.length

        if(tLength < sLength){
            return false
        }
        var startIndex = 0
        for (i in 0..sLength - 1) {
            var isFound = false

            for (j in startIndex..tLength - 1) {

                if(s[i] == t[j]) {
                    startIndex = j + 1
                    isFound = true
                    break;
                }
            }
            if(!isFound){
                return false
            }
        }

        return true
    }
}